app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,68 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+
+  def index
+    if current_user
+    @posts = current_user.posts
+    else
+    @posts = Post.all
+    end
+  end
+
+  def all_posts
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+    @category = Category.new
+  end
+
+  def create
+    @post = current_user.posts.create!(post_params)
+    @category = @post.categories.create!(:category)
```

Comment:
Do you necessarily want to create a new category every time you create a new post? What if the category already exists? A tag might be more appropriate here...